idaes/power_generation/unit_models/boiler_heat_exchanger.py

```diff
@@ -78,135 +92,279 @@ class DeltaTMethod(Enum):
     coCurrent = 1
 
 
+def delta_temperature_lmtd_callback(b):
+    """
+    This is a callback for a temperature difference expression to calculate
+    :math:`\Delta T` in the heat exchanger model using log-mean temperature
+    difference (LMTD).  It can be supplied to "delta_temperature_callback"
+    HeatExchanger configuration option.
+    """
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return (dT1[t] - dT2[t]) / log(dT1[t] / dT2[t])
+
+
+def delta_temperature_amtd_callback(b):
+    """
+    This is a callback for a temperature difference expression to calculate
+    :math:`\Delta T` in the heat exchanger model using arithmetic-mean
+    temperature difference (AMTD).  It can be supplied to
+    "delta_temperature_callback" HeatExchanger configuration option.
+    """
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return (dT1[t] + dT2[t]) * 0.5
+
+
+def delta_temperature_underwood_callback(b):
+    """
+    This is a callback for a temperature difference expression to calculate
+    :math:`\Delta T` in the heat exchanger model using log-mean temperature
+    difference (LMTD) approximation given by Underwood (1970).  It can be
+    supplied to "delta_temperature_callback" HeatExchanger configuration option.
+    This uses a cube root function that works with negative numbers returning
+    the real negative root. This should always evaluate successfully.
+    """
+    dT1 = b.deltaT_1
+    dT2 = b.deltaT_2
+    temp_units = pyunits.get_units(dT1)
+
+    # external function that ruturns the real root, for the cuberoot of negitive
+    # numbers, so it will return without error for positive and negitive dT.
+    b.cbrt = ExternalFunction(
+        library=functions_lib(), function="cbrt", arg_units=[temp_units]
+    )
+
+    @b.Expression(b.flowsheet().time)
+    def delta_temperature(b, t):
+        return ((b.cbrt(dT1[t]) + b.cbrt(dT2[t])) / 2.0) ** 3 * temp_units
+
+
+def delta_temperature_underwood_tune_callback(b):
```

Comment:
First - do we actually have a use case for this method? The original paper that came up with this "tuned" Underwood approximation misunderstood the original purpose of the function and the meaning of he parameters. And their "tuned" parameters were barely any different from the original Underwood parameters making me fairly certain that their tuning was an overfit to their specific data.

My first impression would be that we should either not support this, or if we do, that the default parameter values should be those of Underwood (and definitely not the "tuned" paper).

Reply:
Really the only reason I kept it was that was what was originally in this model, so I didn't want to mess anything else up.  We probably don't want to change the default here until we search out and update other models that use it.